src/client_impl.cc

```diff
@@ -68,7 +85,7 @@ void MixerClientImpl::Check(const Attributes &attributes, DoneFunc on_done) {
 
   converter_.Convert(attributes, request.mutable_attributes());
   request.set_global_word_count(converter_.global_word_count());
-  request.set_deduplication_id(std::to_string(deduplication_id_++));
+  request.set_deduplication_id(GenerateUUID());
```

Comment:
How fast does this end up being? If generating a UUID in this context can cost many cycles, you could consider generating a single UUID at the start and then just taking on a monotonically incrementing counter to the end of the UUID. This would produce a unique string.